packages/api/test/e2e/api.create.e2e.ts

```diff
@@ -37,4 +37,22 @@ describe('e2e api create', () => {
         (websocket as any).websocket.onclose = null;
         (websocket as any).websocket.close();
     });
+
+    it('should create an Api instance with the timeout option', async () => {
+        const endPoint = 'wss://kauri.centrality.cloud/ws?apikey=d449e2d0-868a-4f38-b977-b99e1476b7f0'
+        const api = await Api.create({provider: endPoint, timeout: 1000000000});
+        const hash = await api.rpc.chain.getBlockHash() as Hash;
+
+        expect(hash).toBeDefined();
+    });
+
+    it('should get rejected if the connection fails', async () => {
+        const incorrectEndPoint = 'wss://kauri.centrality.cloud/'
+        await expect(Api.create({provider: incorrectEndPoint})).rejects.toBeDefined();
+    });
+
+    it('should get rejected if it is not resolved in a specific period of time', async () => {
+        const endPoint = 'wss://kauri.centrality.cloud/ws?apikey=d449e2d0-868a-4f38-b977-b99e1476b7f0'
+        await expect(Api.create({provider: endPoint, timeout: 1})).rejects.toBeDefined();
```

Comment:
use `toThrow('specified messages')`

Reply:
used reject.toBeDefined because toThrow only work when the rejected value is an Error object. WebSocket connection fail returns an Event object by default. I will update Api to return an Error and using toThrow here